src/Tasks/Microsoft.NET.Build.Tasks/ResolvePublishAssemblies.cs

```diff
@@ -45,12 +47,20 @@ protected override void ExecuteCore()
             IEnumerable<string> privateAssetsPackageIds = PackageReferenceConverter.GetPackageIds(PrivateAssetsPackageReferences);
             IPackageResolver packageResolver = NuGetPackageResolver.CreateResolver(lockFile, ProjectPath);
 
+            LockFile FilterLockFile = null;
+            if (!string.IsNullOrEmpty(FilterProjectAssetsFile))
+            {
+                FilterLockFile = new LockFileCache(BuildEngine4).GetLockFile(FilterProjectAssetsFile);
```

Comment:
Can reuse the LockFileCache created above, so we don't need to create 2 instances of this object in the same method.